### ISSUE TITLE
Why can't leela zero break the net?

### ISSUE THREAD

**Issue by chunjiongzhang** [gcp/leela-zero#1801]
Can the contents of leela zero's network file be opened? What to open the file with? Why can't leela zero break the net? What content is uploaded to the server every time the next child is downloaded and what is downloaded? Where can I see it in the source program?

**Comment by cjohnchen** [comment]
what do youo want?

**Comment by Marcin1960** [comment]
@chunjiongzhang 

Net file is just a plain text file with a lot of numbers. You can read them all if you wish :)

**Comment by Ishinoshita** [comment]
From what I have observed:

It writes the data on your local drive, then  will connect to the training server as soon as it completes a game (it need to know the game outcome to add it to training data for each position) and will upload the training data (and remove it from you PC, by the way) , even it continues to  play other self-play games. It there is no internet connection, it will just save the data on your disk.
 It will automatically upload the data the next time you launch autogtp and you have a connection.

**Comment by chunjiongzhang** [comment]
So, is the data uploaded by leela contain  weight， task instruction or a chess game for each game?   Is it necessary to upload the chess piece status or download weights for each step?

**Comment by chunjiongzhang** [comment]
Leela can use the opengo elf weight to play, I want to know how the elf weight is mapped to leela's neural network structure.

**Comment by odeint** [comment]
@Ttl wrote about the weight conversion here https://github.com/gcp/leela-zero/issues/1311#issuecomment-386357702 , his script is here https://gist.github.com/Ttl/93bddf45caa969a6b153b26d80be38fe .

The Leela binary also needed a minor fix because of the way the winrate is given by the ELF weights (always from black's view vs from whoever colour's turn it is, if I recall correctly).

**Comment by odeint** [comment]
I have no idea what BS is or IMPLLAMA. As far as I know, Leela's training isn't distributed at all. Only the generation of the _training data_ is distributed. The actual training happens on single machines owned by @gcp or @bjiyxo

**Comment by gcp** [comment]
I didn't find any reference for IMPLLAMA.

A3C is not too dissimilar from what we do, though it has the clients calculating the gradients for the network for their own training data.

**Comment by chunjiongzhang** [comment]
thank u  for  your answer，i got it

**Comment by kennyfs** [comment]
the net will be in autogtp/networks/
